### ISSUE TITLE
Failure to properly escape backslash

### ISSUE THREAD

**Issue by DNGros** [xflr6/graphviz#80]
Thanks for your work putting this library together. It is really useful!

I noticed it looks like backslash escaping is not properly handled. For example:
```python
import graphviz
print(graphviz.__version__)
dot = graphviz.Graph()
dot.node("a", label="foo")
dot.node("b", label='\\')
dot.edge("a", "b")
print(dot)
```
Outputs
```
0.11.1
graph {
	a [label=foo]
	b [label="\"]
	a -- b
}
```

This will not render as the backslash gets treated as a escape of the quote. Expected output is:
```
0.11.1
graph {
	a [label=foo]
	b [label="\\"]
	a -- b
}
```

Other failure cases:
```python
dot.node("c", label="bar\"")
dot.node("d", label=r'"\"')
```
```
graph {
	c [label="bar\""]
	d [label="\"\\""]
}
```

I'm running python 3.6.5.

**Comment by xflr6** [comment]
Thanks. Right, if quotes are escaped with backslashes, literal backslashes should normally be escaped.

The [DOT lang docs](https://graphviz.gitlab.io/_pages/doc/info/lang.html) might be misleading here (or  the  escape interpretation is done by the Layout engine?):
> An ID is one of the following: 
> (...)
> any double-quoted string ("...") possibly containing escaped quotes (\")1; 
> (...)
> 1. In quoted strings in DOT, the only escaped character is double-quote (").
> That is, in quoted strings, the dyad \\" is converted to "; all other characters are left unchanged.
> In particular, \\\\ remains \\\\. Layout engines may apply additional escape sequences.

**Comment by DNGros** [comment]
Hm, I skimmed that page when I encountered the issue, but didn't notice that footnote. You're right. It looks like by the language spec there shouldn't be a need to escape backslashes.

I'm not really familiar with what the layout engine ecosystem looks like, but at least in graphviz 2.40.1 it expects the backslashes to be escaped. This makes sense, as it really complicates parsing if it has potentially seriously backtrack to figure out whether \" means a trailing backslash or the end of the ID, rather than an escaped quote.

Not sure what the best course is here, but it seems like if graphviz is the target backend for this package, the implementation should just match theirs so that it works, even if it conflicts with the language spec. I suppose it could be a configurable option, but not sure how much a use case there is for that though.

**Comment by xflr6** [comment]
Merged the missing escaping of backslashes in a29eaa0bb52fb7a1ffeb8ad6cb05b07b059a064d.

We might want to verify that we get correct results with this change also for the other engines:
https://github.com/xflr6/graphviz/blob/a29eaa0bb52fb7a1ffeb8ad6cb05b07b059a064d/graphviz/backend.py#L21

**Comment by xflr6** [comment]
Seems to work: [Backslash.ipynb](https://github.com/xflr6/graphviz/blob/backslash/Backslash.ipynb)  (`graphviz.version() == (2, 38, 0)`)

**Comment by xflr6** [comment]
Fixed in [0.12 release](https://pypi.org/project/graphviz/0.12/#files):  (00e5eccaf5f986995b9bf81464a6b338541317b2)
